scripts/typescript/src/bcdb/bcdb.ts

```diff
@@ -139,242 +155,304 @@ export class BCDBLoader {
+            unit: null,
+            citation,
+          });
+          this.validate_and_push_property( block2, {
+            node: ["Property"],
+            key: "conc_vol_fraction",
+            type: 'value',
+            value: row[Column.f2],
+            method: row[Column.f2_method],
+            unit: null,
+            citation,
+          });
+          this.validate_and_push_property( block2, {
+            node: ["Property"],
+            key: "conc_vol_fraction",
+            type: 'value',
+            value: row[Column.ftot2],
+            method: row[Column.ftot2_method],
+            unit: null,
+            citation,
+          });
+          this.validate_and_push_property( block2, {
+            node: ["Property"],
+            key: "conc_mass_fraction",
+            type: 'value',
+            value: row[Column.w2],
+            method: row[Column.w2_method],
+            unit: null,
+            citation,
+          })
+          this.validate_and_push_property( block2, {
+            node: ["Property"],
+            key: "density",
+            type: 'value',
+            value: row[Column.rho2],
+            method: row[Column.rho2_method],
+            unit: "g/mL",
+            citation,
+          })
+          this.validator.validate_or_throw(block2);
+          project.material.push(block2);
 
+          this.logger.info(`Add blocks to process as ingredients ...`);
           polymer.component = [block1, block2];
+        };
+    }
+    this.logger.prefix = null;
 
-          //-- add meta data for each material
-          project.material.forEach((material) => {
-            if (!material.bigsmiles) return log(`No bigsmiles found for ${material.name}`);
-
-            // try to get meta data from blocks map
-            // meta data are coming from the 'blocks' tab in the xlsx
-            const metaData = blocks.get(material.bigsmiles);
-            if (!metaData) return error(index, `No meta data found (in blocks) for ${material.bigsmiles}, keep name: ${material.name} `);
-            polymer.name = metaData.name;
-            polymer.names = [metaData.polymer];
-            //log(`Meta data found, name and names[] added`);
-          });
-        });
+    //-- add meta data for each material
+    for(const material of project.material) {
+      if (!material.bigsmiles) {
+        this.logger.info(`No bigsmiles found for ${material.name}`);
+        continue;
+      } 
+
+      // try to get meta data from blocks map
+      // meta data are coming from the 'blocks' tab in the xlsx
+      const metaData = blocks.get(material.bigsmiles);
+      if (!metaData) {
+        this.logger.warning(`No meta data found (in blocks) for '${material.bigsmiles}', keep name: ${material.name} `);
+        continue;
+      }
+      material.name = metaData.name;
+      material.names = [metaData.polymer];
+      this.logger.debug(`Meta data found for '${material.bigsmiles}', name and names[] added`);
+    };
```

Comment:
Because it was unnecessary to run this check until all materials have been generated. It was causing a huge computation and memory cost.